nilearn/_utils/data_gen.py

```diff
@@ -744,6 +748,21 @@ def basic_confounds(length, random_state=0):
     return confounds
 
 
+def add_metadata_to_bids_derivatives(bids_path: str | Path,
+                                      metadata: dict,
+                                      json_file: str = None) -> Path:
```

Comment:
The spacing looks off here. Any clue why flake8 didn't catch it?

```suggestion
def add_metadata_to_bids_derivatives(bids_path: str | Path,
                                     metadata: dict,
                                     json_file: str = None) -> Path:
```

Reply:
Also, shouldn't `black` reformat it to something like the following?

```suggestion
def add_metadata_to_bids_derivatives(
    bids_path: str | Path,
    metadata: dict,
    json_file: str = None,
) -> Path:
```

Reply:
that's because flake8 and black are not yet active on those files

trying to keep a list of things that are covered here: https://github.com/nilearn/nilearn/projects/8#card-88537577

Otherwise check the long exception list in here to see what black works or does not work on.

https://github.com/nilearn/nilearn/blob/a744d100c52d6844f643343797eb36d08e04470d/pyproject.toml

so while reformatting is happening, expect some more "liberal" formatting.